src/xkb_bell.c

```diff
@@ -76,8 +79,10 @@ void initialize_xkb(void) {
 		return;
 	}
 
-	if (! XkbOpenDisplay(DisplayString(dpy), &xkb_base_event_type, &ir,
-	    NULL, NULL, &reason) ) {
+	xkb_disp = XkbOpenDisplay(DisplayString(dpy), &xkb_base_event_type, &ir,
+		NULL, NULL, &reason);
+
+	if (!xkb_disp) {
```

Comment:
maybe this should be refactored to simply use https://linux.die.net/man/3/xkbqueryextension @The-42 ?